src/test/StkAaveRetrievalE2E.t.sol

```diff
@@ -0,0 +1,181 @@
+        assertEq(proposalPayloadAddress, proposal.targets[0], "TARGET_IS_NOT_PAYLOAD");
+
+        IAaveGovernanceV2.ProposalState state = aaveGovernanceV2.getProposalState(proposalId);
+        assertEq(uint256(state), uint256(IAaveGovernanceV2.ProposalState.Queued), "PROPOSAL_NOT_IN_EXPECTED_STATE");
+    }
+
+    function testClaimerUnsetPreProposal() public {    
+        // Call retrieve() on StkAaveRetrieval contract as Balancer multisig without the Proposal being executed
+        vm.prank(balancerMultisig);
+        vm.expectRevert(bytes("Contract not set as claimer"));
+        stkAaveRetrieval.retrieve();
+        
+    }
+
+    function testRetrieveNotBalancerMultisigPreProposal() public {
+        // Call retrieve() on StkAaveRetrieval contract without the Proposal being executed
+        // the msg.sender will be this contract, which is NOT the balancer multisig
+        _executeProposal();
+        vm.expectRevert(bytes("Only Balancer Multisig"));
+        stkAaveRetrieval.retrieve();
+    }
+
+    function testExecute() public {
+        // check that originally claimer is unset, then is correctly set to retrieval contract
+        IAaveIncentivesController incentivesController = IAaveIncentivesController(incentivesControllerAddr);
+        assertEq(incentivesController.getClaimer(balancerDAO), address(0), "CLAIMER_NOT_ZERO");
+        _executeProposal();
+        assertEq(incentivesController.getClaimer(balancerDAO), address(stkAaveRetrieval), "CLAIMER_NOT_RETRIEVAL_CONTRACT");
+    }
+
+    function testRetrievePostProposal() public {
+        _executeProposal();
+
+        // check stkAaveBalance of balancermultisig is zero 
+        assertEq(STK_AAVE.balanceOf(balancerMultisig), 0, "BALANCER_MULTISIG_STK_AAVE_BALANCE_NOT_ZERO");
+        
+        // check initial aToken reward balances of Balancer DAO
+        uint256 expected_balance_ADAI = 507223394753535214703;
+        uint256 expected_balance_AUSDC = 601800814685127542984;
+        uint256 expected_balance_AUSDT = 390825845735555687901;
+        assertEq(wrapped_aDAI.getUnclaimedRewards(balancerDAO), expected_balance_ADAI, "INCORRECT_WRAPPED_ADAI_REWARDS_INITIAL_BALANCE");
+        assertEq(wrapped_aUSDC.getUnclaimedRewards(balancerDAO), expected_balance_AUSDC, "INCORRECT_WRAPPED_AUSDC_REWARDS_INITIAL_BALANCE");
+        assertEq(wrapped_aUSDT.getUnclaimedRewards(balancerDAO), expected_balance_AUSDT, "INCORRECT_WRAPPED_AUSDT_REWARDS_INITIAL_BALANCE");
+
+        // Mock as Balancer Multisig and call retrieve() on StkAaveRetrieval contract
+        vm.prank(balancerMultisig);
+        stkAaveRetrieval.retrieve();
+        
+        // check that stkAave balance is correct now 
+        uint256 balancerDAOStkAAVEBalance = STK_AAVE.balanceOf(balancerMultisig);
+        uint256 expectedStkAaveBalance = 1499850055174218445588;
+        assertEq(balancerDAOStkAAVEBalance, expectedStkAaveBalance, "BALANCER_MULTISIG_STK_AAVE_BALANCE_NOT_ZERO");
```

Comment:
This set of checks seem weird (i.e as in using the absolute values). Is there a better way of checking the before and after balances after retrieval?

Reply:
I can compare the before balance and after balance. i figured it might've been better to have a discrete amount but either will pass